src/read_next/read_next.jsx

```diff
@@ -1,59 +1,45 @@
-import React from 'react';
-import './read_next.css';
-import { Link } from 'react-router-dom';
-const links = [{
-    src: "/photos/1.jpg",
-    alt: "Granny gives everyone the finger",
-    text: "Granny gives everyone the finger, and other tips from OFFF Barcelona",
-},
-{
-    src: "/photos/2.jpg",
-    alt: "Hello world",
-    text: "Hello world, or, in other words, why this blog exists",
-},
-{
-    src: "/photos/3.jpg",
-    alt: "regarding how wework",
-    text: "Here are some things you should know regarding how we work ",
-},
-{
-    src: "/photos/4.jpg",
-    alt: "remote work",
-    text: "Updating list of 50+ sources on distributed teams, remote work, and how to make it all",
-},
-{
-    src: "/photos/5.jpg",
-    alt: "A few words about this blog platform",
-    text: "A few words about this blog platform, Ghost, and how this site was made",
-},
-{
-    src: "/photos/6.jpg",
-    alt: "modern remote working",
-    text: "How modern remote working tools get along with Old School Cowboy's methods",
-},
-];
-const linksList = links.map(function (link, index) {
-    return <Link to={"/posts/" + index} className='decoration'> <div className="atribute">
-        <img className="second_page_photos" width="304" height="176" src={link.src}
-            alt={link.alt} />
-        <div>
-            <p className="text_after_images">{link.text}</p>
-        </div>
-    </div></Link>;
-})
-class ReadNext extends React.Component {
-    constructor(props) {
-        super(props);
-    }
-    render() {
-        return (
-            <section className="additional_articles">
-                <h2 className="read_next">What to read next</h2>
-                <div className="photo">
-                    {linksList}
-                </div>
-            </section>
-        );
-    }
+import React, { useEffect, useState } from "react";
+import "./read_next.css";
+import { Link, useParams } from "react-router-dom";
+
+const ReadNext = () => {
+    const params = useParams();
+    const current = params.id;
+    const [data, setPosts] = useState({ posts: [] });
+    useEffect(() => {
+        fetch("https://dolphin-app-cbjj4.ondigitalocean.app/users/misha/posts")
+            .then((response) => response.json())
+            .then((data) => setPosts({ posts: data }));
+        const correcterArr = data.posts.filter((elem) => {
+            if (elem.id !== parseInt(current)) {
+                return elem;
+            }
+        });
+        setPosts({ posts: correcterArr });
+    }, [params]);
```

Comment:
1. please read again with much attention about Promises and useEffect dependencies. you're using it wrong. for example here your useEffect depends on `current` variable, but you pass `params` variable to the dependencies array. even thought they're connected you need to pass the exact variables that your useEffect depend on 
2. `filter` array method expects a [predicate function](https://dev.to/deogadkarravina/javascript-predicate-and-uses-8pl), it should return true/false. it can work with values that evaluate to booleans, but it's better to adhere to best practices
3. you don't need to parse `params.id` on every step of filter iteration, you can do it outside of the loop

```suggestion
    useEffect(() => {
        fetch("https://dolphin-app-cbjj4.ondigitalocean.app/users/misha/posts")
            .then((response) => response.json())
            .then((json) => {
                      const currentPostId = parseInt(params.id)
                      const postsWithoutCurrent = json.posts.filter((post) => {
                          return post.id !== currentPostId;
                      });
                      setPosts({ posts: postsWithoutCurrent });
            });
    }, [params]);
```